core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2266,7 +2261,7 @@ private void analyzeWindowFunctions(QuerySpecification node, List<Expression> ou
                 ResolvedWindow window = analysis.getWindow(windowFunction);
                 // TODO get function requirements from window function metadata when we have it
                 String name = windowFunction.getName().toString().toLowerCase(ENGLISH);
-                if (name.equals("lag") || name.equals("lead")) {
+                if ("lag".equals(name) || "lead".equals(name)) {
```

Comment:
We don't have convention either way, and the `name` cannot be null, so this change isn't necessary